api/circulation.py

```diff
@@ -568,11 +569,6 @@ def borrow(self, patron, pin, licensepool, delivery_mechanism,
 
         internal_format = api.internal_format(delivery_mechanism)
 
-        if patron.fines:
-            max_fines = Configuration.max_outstanding_fines(patron.library)
-            if max_fines is not None and patron.fines > max_fines.amount:
-                raise OutstandingFines()
-
```

Comment:
This code is redundant with what happens inside PatronUtility.assert_borrowing_privileges, and it has the same bug that assert_borrowing_privileges used to have, so the best thing is to remove it.